src/Amazon.Lambda.Tools/TemplateProcessor/UpdatableResource.cs

```diff
@@ -140,13 +141,44 @@ public bool IsCode
                     {
                         return false;
                     }
-                    if(!string.IsNullOrEmpty(this._resource.DataSource.GetValue("Code", "ZipFile")))
+                    if (!string.IsNullOrEmpty(this._resource.DataSource.GetValue("Code", "ZipFile")))
                     {
                         // The template contains embedded code.
                         return false;
-                    }                     
+                    }
+                    else
+                    {
+                        string localPath = GetLocalPath();
```

Comment:
I find this logic hard to follow. In particular comparing `localPath` with `tempLocalPath` and what is a `tempLocalPath`. What about something like this.

```
string localPath = this._resource.DataSource.GetValueFromResource(LambdaConstants.CF_SERVERLESS_METADATA, LambdaConstants.CF_SERVERLESS_DOCKERCONTEXT);
// Pointing to already pushed image, no package needs to be done.
if (IsECRImage(localPath))
{
    return false;
}
// metadata points to local path that needs to be packaged up return true for code.
else if (!string.IsNullOrEmpty(localPath))
{
    return true;
}

// If no docker metadata was set fallback to looking for a local path in the resource's ImageUri property.
localPath = this._resource.DataSource.GetValue(LambdaConstants.CF_LAMBDA_IMAGEURI);
// Pointing to already pushed image, no package needs to be done.
if (IsECRImage(localPath))
{
    return false;
}

return true;
```

Reply:
Done